Fix for DTL status colors on vDisk, vMachine and vPool

kvanhijf opened the pull request.

kvanhijf committed:
Fix for DTL status colors on vDisk, vMachine and vPool

kvanhijf committed:
Fixed unittests

khenderick commented on ovs/dal/hybrids/vdisk.py:
```diff
@@ -76,6 +78,36 @@ def storagedriver_client(self):
+        """
+        Retrieve the DTL status for a vDisk
+        """
+        sd_status = self.info.get('failover_mode', 'OK_STANDALONE').lower()
```
Shouldn't we work with an `unknown` status here?

khenderick commented on ovs/dal/hybrids/vdisk.py:
```diff
@@ -76,6 +78,36 @@ def storagedriver_client(self):
+        possible_dtl_targets = set()
+        for sr in StorageRouterList.get_storagerouters():
+            if sr.guid == self.storagerouter_guid:
+                domains = [junction.domain for junction in sr.domains if junction.backup is False]
```
The if should be removed

khenderick commented on ovs/dal/hybrids/vpool.py:
```diff
@@ -65,6 +66,13 @@ def storagedriver_client(self):
+        VPool configuration
+        """
+        from ovs.lib.vpool import VPoolController
+        return VPoolController.get_configuration(self.guid)
```
Don't use controllers in the DAL, it mixes up our n-tier architecture.

kvanhijf committed:
Fixes based on review
Added additional unittest for DTL
Fixed issue for DTL configuration when going from single node with vDisks to multinode
Navigate to vMachine or vDisk overview page when the detail page of a vMachine or vDisk is no longer reachable

khenderick commented on webapps/api/backend/views/vpools.py:
```diff
@@ -171,5 +171,11 @@ def get_configuration(self, vpool):
         """
+        if version > 3:
+            raise NotAcceptable('Only available in API versions smaller than 3')
         return VPoolController.get_configuration.delay(vpool_guid=vpool.guid)
```
You still have to adapt this line

khenderick commented on ovs/lib/storagerouter.py:
```diff
@@ -873,14 +877,14 @@ def add_vpool(parameters):
-            cache_strategy = current_storage_driver_config['cache_strategy']
-            tlog_multiplier = current_storage_driver_config['tlog_multiplier']
-            sco_factor = float(current_storage_driver_config['write_buffer']) / tlog_multiplier / sco_size
+            sco_size = vpool.configuration['sco_size']
```
Can you load this only once, this introduces quite a few memcache hits.

kvanhijf committed:
Changes based on review
